src/plugins/kibana_utils/public/errors/errors.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* eslint-disable max-classes-per-file */
+
+const canStack = (function() {
+  const err = new Error();
+  return !!err.stack;
+})();
+
+// abstract error class
+export class KbnError extends Error {
+  constructor(msg: string, constructor?: Function) {
+    super();
+    this.message = msg;
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, constructor || KbnError);
+    } else if (canStack) {
+      this.stack = new Error().stack;
+    } else {
+      this.stack = '';
+    }
+  }
+}
+
+/**
+ * when a mapping already exists for a field the user is attempting to add
+ * @param {String} name - the field name
+ */
+export class DuplicateField extends KbnError {
```

Comment:
optional: KbnError previously had an abstract public method `displayToScreen()` since only VislibErrors use this, I like that you moved it to the `VislibError` class. But I wonder is there any value in having a `KbnError` class at all, can't `DuplicateField` and `SavedObjectNotFound`, `VislibError` etc just extend from Error?

Why didn't you move `DuplicateField`, `SavedObjectNotFound` & `InvalidJSONProperty` into `public/saved_objects` the same way you moved other errors next to the domain that throws/uses them?

Reply:
First of all, I think that having a common base class is useful. Removing it it will be easy, but adding it back will be hard.

Regarding the 3 errors I moved with the base class (and not to their matching components): as the goal of this task was to remove a ui/public dependency from [data plugin](https://github.com/elastic/kibana/issues/44377), If I had moved them to `ui/saved_objects`, I would still be blocked on them. 

We will move those to `saved_objects` once they are migrated \ shimmed as well.

Reply:
ok, that makes sense